src/core/store.ts

```diff
@@ -26,20 +26,65 @@ type Revision = number
 type InvalidatedRevision = number
 type ReadDependencies = Map<AnyAtom, Revision>
 
-// immutable atom state
+/**
+ * Immutable atom state, tracked for both mounted and unmounted atoms in a store.
+ */
 export type AtomState<Value = unknown> = {
+  /**
+   * Counts number of times atom has actually changed or recomputed.
+   */
   r: Revision
+  /**
+   * Marks the revision of this atom when a transitive dependency was invalidated.
+   * Mounted atoms are considered invalid when `r === i`. Unmounted atoms are
+   * always considered invalid.
```

Comment:
Unmounted atoms are not *always* considered invalid.
The validity will be checked when they are re-mounted.

Reply:
> Unmounted atoms are not always considered invalid.

This is still to be taken into account?

Reply:
I have changed the text to:

```
  * Marks the revision of this atom when a transitive dependency was invalidated.
   * Mounted atoms are considered invalid when `r === i`. Unmounted atoms are
   * considered invalid until they are mounted again.
```

Is that correct?

Reply:
> Unmounted atoms are considered invalid until they are mounted again.

Oh, I see the change now. I meant, unmounted atoms are not considered invalid, the invalidity will be checked after re-mount, and we then know if it's valid or invalid. So, unmounted atoms can still be valid. (Some of tests cover this behavior.)

Reply:
I see. I got the impression from this code:

https://github.com/justjake/jotai/blob/38c1d07616c564b5132af631f563b92f3d6e6526/src/core/store.ts#L485-L488

```
            if (!mountedMap.has(a)) {
              // not mounted
              readAtomState(version, a)
            } else {
```

because we always `readAtomState(version, a)` for unmounted atoms, but for mounted atoms, we only `readAtomState(version, a)` if 

```
              const aState = getAtomState(version, a)
              if (
                aState &&
                aState.r === aState.i // revision is invalidated
              ) {
                readAtomState(version, a)
              }
```

Reply:
It seems like this case is hit when an unmounted atom is read for the first time -- we assume it's invalid?

Reply:
> It seems like this case is hit when an unmounted atom is read for the first time -- we assume it's invalid?

Yeah, that sounds right. We may want to improve the `not mounted` comment.

Reply:
Ah, I see. It's not mounted, so we need check the validity, thus invoking readAtomState.

Reply:
Putting the invalidation algorithm into a comment is too difficult. I removed the note about unmounted atoms here entirely, and tried to add a few more comments to `readAtomState` instead.